.github/workflows/main.yaml

```diff
@@ -6,59 +6,48 @@ name: Build and test linkml-model
 
 on:
   pull_request:
-    branches: [ main ]
-    types: [ opened, synchronize, reopened ]
+    branches: [main]
+    types: [opened, synchronize, reopened]
 
 jobs:
   test:
-
     runs-on: ubuntu-latest
+
     strategy:
       matrix:
         python-version: ["3.9", "3.10"]
 
     steps:
-
       #----------------------------------------------
       #       check-out repo and set-up python
       #----------------------------------------------
       - name: Check out repository
-        uses: actions/checkout@v3
-
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+        uses: actions/checkout@v4.2.2
         with:
-          python-version: ${{ matrix.python-version }}
+          fetch-depth: 0
 
       #----------------------------------------------
       #          install & configure poetry
       #----------------------------------------------
       - name: Install Poetry
-        uses: snok/install-poetry@v1.3
-        with:
-          virtualenvs-create: true
-          virtualenvs-in-project: true
+        run: |
+          pipx install poetry
+          pipx inject poetry poetry-dynamic-versioning
```

Comment:
The repo is not yet setup for dynamic-versioning so you don't need to install it.